src/main/java/com/commercetools/project/sync/CliRunner.java

```diff
@@ -195,16 +196,21 @@ private static Options buildCliOptions() {
   private static CompletionStage<Void> processSyncOptionAndExecute(
       @Nonnull final CommandLine commandLine, @Nonnull final SyncerFactory syncerFactory) {
 
-    final String syncOptionValue = commandLine.getOptionValue(SYNC_MODULE_OPTION_SHORT);
+    final String[] syncOptionValues = commandLine.getOptionValues(SYNC_MODULE_OPTION_SHORT);
     final String runnerNameValue = commandLine.getOptionValue(RUNNER_NAME_OPTION_SHORT);
     final boolean isFullSync = commandLine.hasOption(FULL_SYNC_OPTION_SHORT);
     final boolean isSyncProjectSyncCustomObjects =
         commandLine.hasOption(SYNC_PROJECT_SYNC_CUSTOM_OBJECTS_OPTION_LONG);
 
-    return SYNC_MODULE_OPTION_ALL.equals(syncOptionValue)
-        ? syncerFactory.syncAll(runnerNameValue, isFullSync, isSyncProjectSyncCustomObjects)
-        : syncerFactory.sync(
-            syncOptionValue, runnerNameValue, isFullSync, isSyncProjectSyncCustomObjects);
+    if (SYNC_MODULE_OPTION_ALL.equals(syncOptionValues[0])) {
```

Comment:
I tried moving the validation before processing the sync. But it doesn't look good(Example: collectSyncOptionValues validation is unnecessary when we have only one argument passed). Hence did this changes https://github.com/commercetools/commercetools-project-sync/pull/219/commits/e37d6ebb98005c764ebd0f7251a3af69189bb236 
This will handle all the below cases,

*  -s all, products, categories is not a valid list
* When all is not a first option, Then Multiple Sync method throws an IllegalArgumentException.
* getOptionValues won't be null here, Because we are already handling the empty values case and throwing exception before getting this values.

Reply:
You are not explaining the idea here, I did not say it will not work, I am saying validation is shared in many classes. Also even with your argument, it seems you did some fix according to my comment.

Reply:
I feel this(Currently implemented) approach is better than moving entire validation before the sync. Please recheck once and let me know your view. If you are sure if it is better then I will implement it.